File: 프로그래머스/3/389481. 봉인된 주문/봉인된 주문.java
import java.util.*;
import java.util.stream.Collectors;

class Solution {
    public String solution(long n, String[] bans) {
        String answer = "";
        List<Long> values = Arrays.stream(bans)
            .map(ban -> convertTo26(ban))
            .sorted()
            .collect(Collectors.toList());
        
        var target = n;
        
        for (Long value : values) {
            if (target >= value) target++;
            else break;
        }
        
        return convertToString(target);
    }
    
    String convertToString(long target) {
        var sb = new StringBuilder();
        
        var num = target;
        
        while (num-- > 0) {
            int c = (int)('a' + (num % 26));
            sb.append((char) c);
            num /= 26;
        }
        
        return sb.reverse().toString();
    }
    
    Long convertTo26(String target) {
        // a~z = 26
        Long result = 0L;
        for (char c : target.toCharArray()) {
            int value = c - 'a' + 1;
            result = result * 26 + value;
        }
        
        return result;
    }
}

File: 프로그래머스/3/389481. 봉인된 주문/봉인된 주문.py
def solution(n, bans):
    # a~z = 26개
    def convertTo26(target):
        result = 0
        for c in target:
            result = result * 26 + (ord(c) - ord('a') + 1)
        return result
    
    def convertToString(value):
        num = value
        
        sb = []
        while num > 0:
            num = num - 1
            sb.append(chr(ord('a') + (num % 26)))
            num = num // 26
        return ''.join(reversed(sb))
        
    ban_values = sorted(convertTo26(ban) for ban in bans)
    
    current = n
    for l in ban_values:
        if (current >= l):
            current = current + 1
        else:
            break
        
    return convertToString(current)
